### PR TITLE
feat: impl ajv + typebox Validator

### PR DESCRIPTION
**Pull request by fengmk2**
closes https://github.com/eggjs/tegg/issues/200

## Summary by CodeRabbit

- **New Features**
	- Introduced `@eggjs/ajv-decorator` for enhanced type validation and transformation in TypeScript projects.
	- Added `@eggjs/tegg-ajv-plugin` for parameter validation and type definition in Egg.js applications, with complete TypeScript support.
	- Implemented custom error handling for validation failures in Ajv.
	- Provided installation, configuration, and usage guidelines for `tegg-dal-plugin` in both `egg` and `standalone` modes.

- **Bug Fixes**
	- Removed unnecessary `await` in `Runner.ts` for lifecycle utilities, enhancing performance.

- **Documentation**
	- Updated README files with detailed information on the newly introduced plugins and their usage in projects.
	- Provided comprehensive installation and configuration instructions for the `tegg-dal-plugin`.

- **Tests**
	- Added new tests for AJV validation, ensuring robust error handling and successful data validation.
	- Updated test configurations and imports to reflect changes in the framework.

### CODE REVIEW

File: standalone/standalone/test/index.test.ts [hunk 1]
```diff
@@ -249,4 +249,34 @@
...
+        ]);
+        return true;
+      });
+    });
+
+    it('should pass', async () => {
+      const result = await main<string>(path.join(__dirname, './fixtures/ajv-module-pass'));
+      assert.equal(result, '{"body":{"fullname":"mock fullname","skipDependencies":true,"registryName":"ok"}}');
+    });
+  });
```
[the remark this answers is not in the document]
Comment (fengmk2) [reply standalone/standalone/test/index.test.ts:281]: yes please

File: plugin/ajv/README.md [hunk 2]
```diff
@@ -0,0 +1,639 @@
+# @eggjs/tegg-dal-plugin
```
Comment (fengmk2) [inline plugin/ajv/README.md:1]: 修改

File: standalone/standalone/test/index.test.ts [hunk 3]
```diff
@@ -249,4 +249,42 @@ describe('test/index.test.ts', () => {
       assert.equal(foo.col1, '2333');
     });
   });
+
+  describe('ajv runner', () => {
+    it('should throw AjvInvalidParamError', async () => {
+      await assert.rejects(async () => {
+        await main<string>(path.join(__dirname, './fixtures/ajv-module'), {
+          dependencies: [
+            path.dirname(require.resolve('@eggjs/tegg-ajv-plugin/package.json')),
```
Comment (fengmk2) [inline standalone/standalone/test/index.test.ts:258]: 搞定，又学习了经验

File: core/ajv-decorator/package.json [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  ],
+  "typings": "dist/index.d.ts",
+  "scripts": {
+    "test": "cross-env NODE_ENV=test NODE_OPTIONS='--no-deprecation' mocha",
+    "clean": "tsc -b --clean",
+    "tsc": "npm run clean && tsc -p ./tsconfig.json",
+    "tsc:pub": "npm run clean && tsc -p ./tsconfig.pub.json",
+    "prepublishOnly": "npm run tsc:pub"
+  },
+  "author": "killagu <killa123@126.com>",
```
Comment (killagu) [inline core/ajv-decorator/package.json:25]: fengmk 👀

**Review by killagu — APPROVED** [review]
LGTM

File: core/ajv-decorator/package.json (hunk 4 above)
Comment (fengmk2) [reply core/ajv-decorator/package.json]: 抹掉了。

**Pull request merged**